File: frontend/src/lib/api/pipeline.ts
/*
 1. FILE PURPOSE: API calls to run the evaluation pipeline and poll status.
 2. RESPONSIBILITIES:
    - runPipeline(batchId); getStatus(batchId).
 3. PLANNED CONTENTS: async functions using the shared axios client.
 4. INPUTS / OUTPUTS: Inputs: batch id. Outputs: run ack + status payloads.
 5. DEPENDS ON / USED BY: lib/api/client, types; used by usePipeline.
*/

File: backend/app/api/routes/pipeline.py
# 1. FILE PURPOSE: Exposes the main FastAPI endpoints to trigger the evaluation pipeline (Scan -> Preprocess -> OCR -> Features) and check batch status.
# 2. RESPONSIBILITIES:
#    - POST /run to start processing.
#    - GET /status to poll progress.
#    - Coordinate engine execution.
# 3. PLANNED CONTENTS: FastAPI APIRouter. Maps HTTP requests to background tasks.
# 4. INPUTS / OUTPUTS: Inputs: batch IDs. Outputs: status updates.
# 5. DEPENDS ON / USED BY: FastAPI, pipeline modules.
